### ISSUE TITLE
[New Data Source]: VictoriaMetrics

### ISSUE THREAD

**Issue by hagen1778** [grafana/grafana#91687]
### Background

VictoriaMetrics is a time series database. It can be used for collecting and processing infrastructure metrics, k8s metrics, APM, etc. It is compatible with Prometheus exposition format and protocols.
VictoriaMetrics is open source and distributed under Apache 2.0 License - see https://github.com/VictoriaMetrics/VictoriaMetrics 

### Use case

VictoriaMetrics is widely used in open-source. Especially, in integration with Grafana.
Many users use Prometheus datasource to connect to VictoriaMetrics for querying data in Grafana.
However, VictoriaMetrics provides own datasource plugin - https://github.com/VictoriaMetrics/victoriametrics-datasource

This plugin is open-source and publicly available for everyone. It is developed and supported by VictoriaMetrics maintainers and community members and does not require development efforts from the Grafana team. It would be great to have this plugin signed to make it easier for users to install. Currently, users need to change Grafana's security settings in order to install the plugin.

### Contribution

- [X] Are you looking to create the plugin?
- [X] Are you affiliated with the project/product the data source integrates with?
- [x] Does the plugin integrate with a commercial product?

**Comment by josiahg** [comment]
VictoriaMetrics is a commercial entity and for this data source would fall under our `Commercial` license type.

See Grafana Labs [Plugins Policy](https://grafana.com/legal/plugins/):

```
Commercial plugins are offered in connection with a commercial offering, commercial dependencies or offered by a for-profit business. They contain dependent technologies that are closed source or commercially backed (even if open source at their core).
```

Happy to discuss with the team at VictoriaMetrics - please reach out!

**Comment by matejsp** [comment]
We are using open source version of VictoriaMetrics and grafana in production but are unable to add it to Grafana via UI.
I saw that you are blocking changes to prometheus to support VictoriaMetrics (using the same PromQL) -> https://github.com/grafana/grafana/pull/81436.

Open source plugin from github is unsigned because you want substantial amount of money EVERY year to allow signing EVEN for purely open source setup -> https://github.com/VictoriaMetrics/victoriametrics-datasource/issues/87.

You want to mark open source plugin and open source database as Commerical so you need to buy grafana enterprise even for end users to support enterprise plugin even for FULLY open source stack that we and others use. Yes victoria metrics have also commercial offering with couple of additional feature but most of us use open source version.

I see this as very unopensource like spirit and should be made more visible publicly on slashdot/reddit.

From our testing victoria metrics open source version is times better in comparison to Prometheus and this competes with your Prometheus as a service and you just want to stiff open source competition.

**Comment by nebula-it** [comment]
Hey @josiahg 
Can we please get an update here? 
Thank you

**Comment by StianOvrevage** [comment]
I hate to be negative but the elephant in the room is that VictoriaMetrics (and the recently released [VictoriaLogs](https://docs.victoriametrics.com/victorialogs/) is leaps and bounds ahead of the chaotic convoluted mess that the Grafana stack has (de)volved into the past years.

And it's quite telling that making it as difficult as possible to use VictoriaMetrics (and Logs) in Grafana is the path chosen to advance their own solutions (Loki, Tempo, Mimir, etc). Instead of offering competitive solutions that actually work and can be operated.

I don't say this lightly. I've been a huge fan of Grafana and blogged about it all the way back in 2014.

**Comment by leeoniya** [comment]
IIUC you can simply use the built-in Prometheus datasource with VM, since they aim to be prometheus-compatible. I imagine if they made VictoriaLogs api-compatible with Loki, then the same would be true there.

I don't think there is anything preventing VM or VL plugins from existing in the [plugins marketplace](https://grafana.com/grafana/plugins/), especially if they want to offer features that are beyond or different than what Prometheus/Loki offers.

It's not unreasonable that there would be substantial debate about whether Grafana should include and maintain plugins for a direct commercial competitor, right?

(i work at Grafana, but not on the datasources)

**Comment by nebula-it** [comment]
> IIUC you can simply use the built-in Prometheus datasource with VM, since they aim to be prometheus-compatible. I imagine if they made VictoriaLogs api-compatible with Loki, then the same would be true there.

Yes, VM is Prometheus compatible but [MetricsQL ](https://docs.victoriametrics.com/metricsql/)solves a lot of issues that are still [present ](https://medium.com/@romanhavronenko/victoriametrics-promql-compliance-d4318203f51e)in PromQL. So there is practically no reason to be hamstrung by old standard when a superior solution exists. Additionally, Prometheus has become an industry standard especially in CNCF landscape, whereas Loki is in no such place. And same as MetricsQL, [LogsQL](https://docs.victoriametrics.com/victorialogs/logsql/) offers far better performance and feature set compared to Loki so again I see no reason to hold VLogs back other than Grafana intentionally hindering the progress.


> I don't think there is anything preventing VM or VL plugins from existing in the [plugins marketplace](https://grafana.com/grafana/plugins/), especially if they want to offer features that are beyond or different than what Prometheus/Loki offers.

Unfortunately there is. As you can see in this very thread Grafana plugin policy is very vague (which seems to be intentional) and the lack of response from Grafana pretty much confirms their intentions that they just want to gatekeep for as long as possible.

> It's not unreasonable that there would be substantial debate about whether Grafana should include and maintain plugins for a direct commercial competitor, right?

Given how much Grafana emphasizes about their open sourceness, this is absurd to even think of. Should Microsoft not allow Chrome on Windows? Grafana's home page is filled with quotes like:

> Open source is at the heart of what we do at Grafana Labs.
Open source is in our DNA

Which makes it even worse that one open source project would gate-keep another just for the fear of competitiveness, whereas we as open source community should be helping each other grow and let the technology/solutions speak for themselves rather than hindering innovation.

P.S: I am not affiliated with either Grafana and VictoraMetrics. Just a user of both

**Comment by hagen1778** [comment]
Hello @nebula-it @StianOvrevage ! 
VictoriaLogs datasource plugin was submitted on commercial basis to Grafana marketplace on 20th of December 2024. We're still waiting for a review, though. I believe, once we pass the review the plugin for VictoriaLogs will be available for all Grafana users via marketplace.
Once we proceed with VictoriaLogs datasource, we will submit VictoriaMetrics datasource shortly after.

**Comment by leeoniya** [comment]
> Unfortunately there is. As you can see in this very thread Grafana plugin policy is very vague (which seems to be intentional) and the lack of response from Grafana pretty much confirms their intentions that they just want to gatekeep for as long as possible.

you're right, i missed that. clearly i'm new to this thread and topic :)

i don't have any inside baseball as i'm not involved in any commercial conversations or policy decisions.

i think the main complaint is that Grafana is unwilling to sign plugins that are in conflict with our commercial offerings. that may very well be the case. i'm not sure it's hugely different than Apple or Google or MS expecting a cut of the revenue from apps they approve for their app stores.

> Given how much Grafana emphasizes about their open sourceness, this is absurd to even think of. Should Microsoft not allow Chrome on Windows? Grafana's home page is filled with quotes like:

Grafana is a commercial endeavor built on top of OSS. zero gate-keeping in its hosted services / SaaS platform is a weird thing to expect, and i'm not sure how that violates the nature of "open sourceness".

it's not exactly difficult to install unsigned plugins:
https://github.com/VictoriaMetrics/victoriametrics-datasource#why-victoriametrics-datasource-is-unsigned

anyways, as an OSS dev for far longer than a Grafana dev, i understand your frustration. however, i also know what it's like to spend over a decade writing, maintaining, and supporting MIT code for free that other companies that definitely monetize without paying me anything, i understand the other side as well.

**Comment by vhajdukd** [comment]
Is this review finished ? I've just started up grafana 11.5 and I can see on the list Signed VM ds 0.13.0 ?

**Comment by hagen1778** [comment]
Hey @vhajdukd! [VictoriaMetrics datasource](https://grafana.com/grafana/plugins/victoriametrics-metrics-datasource/) was published yesterday. It still has some rough edges in the readme formatting which are being addressed right now. But it can be installed and used already.

[VictoriaLogs datasource](https://grafana.com/grafana/plugins/victoriametrics-logs-datasource/) is still in review stage.

**Comment by josiahg** [comment]
> Hey [@josiahg](https://github.com/josiahg) Can we please get an update here? Thank you

hey just seeing this - I left Grafana Labs in October 2024, so I am no longer in a position to comment further on this.

**Comment by nebula-it** [comment]
All good, looks like its resolved now and VM datasource if official now.

**Comment by hagen1778** [comment]
VictoriaMetrics plugins for metrics and logs are now publicly available to everyone:
* https://grafana.com/grafana/plugins/victoriametrics-metrics-datasource; [github repo](https://github.com/VictoriaMetrics/victoriametrics-datasource)
* https://grafana.com/grafana/plugins/victoriametrics-logs-datasource; [github repo](https://github.com/VictoriaMetrics/victorialogs-datasource)

If you have any issues with plugins, please open a ticket in the corresponding github repo.
Thanks to all!
